plasmapy/__init__.py

```diff
@@ -41,11 +41,37 @@
 import pkg_resources
 
 try:
-    # ./version.py is generated by setuptools_scm during install
-    __version__ = pkg.resources.get_distribution("plasmapy").version
-except Exception:
-    # setuptools_scm couldn't generate __version__; probably package installed from source?
-    __version__ = "unknown"
+    # this places a runtime dependency on setuptools
+    #
+    # note: if there's any distribution metadata in your source files, then this
+    #       will find a version based on those files.  Keep distribution metadata
+    #       out of your repository unless you've intentially installed using
+    #       as editable (e.g. `pip install -e `)
+    __version__ = pkg_resources.get_distribution("plasmapy").version
+except pkg_resources.DistributionNotFound:
+    # package is not installed
+    fallback_version = 'unknown'
+    try:
+        # code most likely being used from source
+        # if setuptools_scm is installed then generate a version
+        from setuptools_scm import get_version
+        __version__ = get_version(root='..',
+                                  relative_to=__file__,
+                                  fallback_version=fallback_version)
+        del get_version
+        warn_add = 'setuptools_scm failed to detect the version'
+    except ModuleNotFoundError:
+        # setuptools_scm is not installed
+        __version__ = fallback_version
+        warn_add = 'setuptools_scm is not installed'
+
```

Comment:
```suggestion
   warn_add += ". Consider using `pip install -e {plasmapy_directory_root}`"
```

Reply:
I don't think I want to encourage this because the version info is only generated during the install, so `__version__` won't get updated with each successive commit.

Reply:
How would you setup a dev environment, then?

Reply:
I use PyCharm, so I'm basically sitting in the root of the repository.  If we'd want a continuously updated `__version__` in the editable package case, then `setuptools_scm` would have to be used.  However, I don't know if there's a way to indicated if a package is installed as editable.

Reply:
Do we really need a continuously up-to-date version attribute, though? Seriously. I struggle to think of practical use cases for this. Besides, if you really need the version tag to update, you can run pip install again and that'll trigger the version rebuild.

Reply:
No, I don't think we need `__version__` to be continuously update, that's why I added the note in the comments about this.  If some unforeseen scenario arises, then the information is at least there.

The try-statement will basically cover any distribution installed with `pip`.  The `setuptools_scm` in the except-statement, will cover programmers like me who work in the repository's root directory instead of installing.  It's likely there will not very many use cases for people like me.  The only one I can think of off the top of my head, is checking the `__version__` is properly incremented when the commit tag is defined.

Of course, I did discover this root repository metadata thing when I was getting a defined `__version__` when I knew there shouldn't be one.  This was because I still had metadata in the root directory from a previous editable install and the uninstall didn't remove the metadata.